pkg/build/controller/build/build_controller.go

```diff
@@ -258,15 +259,19 @@ func shouldIgnore(build *buildapi.Build) bool {
 	}
 
 	// If a build is in a terminal state, ignore it; unless it is in a succeeded or failed
-	// state and its completion time is not set, then we should at least attempt to set its
-	// completion time if possible.
+	// state and its completion time or outputexcerpt is not set, then we should at least attempt to set its
```

Comment:
Is it actually the case that we enter here twice, once on becoming aware that the build has failed and a second time when the excerpt text becomes available? (and that the order of those events is not reliable?)

If that's not the case (which I hope), then to avoid complicating matters, if possible I think we should keep the gating condition as simple as possible - e.g. `build.Status.CompletionTimestamp == nil` only - and the controller should set the OutputExcerpt as part of the update that sets CompletionTimestamp.

I would very much like to see the builder pod no longer updating Status.Phase, so that we can factor out this special case - avoiding complicating it in the meantime is a plus.